core/src/math/Base58.cpp

```diff
@@ -35,34 +35,44 @@
 #include <crypto/HashAlgorithm.h>
 #include <utils/hex.h>
 #include <functional>
-#include <utils/hex.h>
 #include <crypto/Keccak.h>
 
 static const std::string DIGITS = "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz";
+static const std::string RIPPLE_DIGITS = "rpshnaf39wBUDNEGHJKLM4PQRST7VWXYZ2bcdeCg65jkm8oFqi1tuvAxyz";
 static const ledger::core::BigInt V_58(58);
 
-static void _encode(const ledger::core::BigInt& v, std::stringstream& ss) {
+static std::string getBase58Dictionary(const std::string &networkIdentifier) {
```

Comment:
It depends how we look at it, because first it is specific to Base58 encoding so it can have it's place in this file, second if we follow same logic we should also get rid of EIP55 encoding which is specific to ETH ! 
What we can do when re-architecturing libcore is having specific implementations per coin family belonging to this family's module

Reply:
And where this function would live ? In same file ?

Reply:
Ok, after a live discussion, I got why you did that. LGTM for me then.